cmd/collection/main.go

```diff
@@ -364,13 +364,17 @@ func main() {
 				return nil, err
 			}
 
+			createMetrics := func(chainID flow.ChainID) module.HotstuffMetrics {
```

Comment:
Make metrics constructor configurable to avoid multiple test nodes registering the same metric name multiple times with the metrics library singleton.

Reply:
The test node and all its immediate dependencies get passed a `NoopCollector`, however HotStuff is instantiated on a per-epoch basis with a different cluster chain ID. 

Previously we would use [`metrics.NewHotStuffCollector`](https://github.com/onflow/flow-go/blob/master/engine/collection/epochmgr/factories/hotstuff.go#L66) within the `HotStuffFactory`. In other words, even if the test node was using a noop metrics collector, the factory would still create a "real" metrics collector for HotStuff. Since we are running multiple nodes in the same process, this caused errors in the metrics library registration singleton. 

Passing in `createMetrics` lets us configure the `HotStuffFactory` to create noop metrics for tests.